helm/cluster-vsphere/files/opt/bin/set-hostname.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+set -x
+echo "${COREOS_CUSTOM_HOSTNAME}" > /etc/hostname
+hostname "${COREOS_CUSTOM_HOSTNAME}"
+echo "::1         ipv6-localhost ipv6-loopback" >/etc/hosts
+echo "127.0.0.1   localhost" >>/etc/hosts
+echo "127.0.0.1   ${COREOS_CUSTOM_HOSTNAME}" >>/etc/hosts
```

Comment:
moved out of the [ignition spec](https://github.com/giantswarm/cluster-vsphere/blob/main/helm/cluster-vsphere/templates/_helpers.tpl#L269-L275)